src/server/pps/server/master.go

```diff
@@ -163,10 +163,7 @@ eventLoop:
 				if pc, ok := m.pcMgr.pcs[e.pipeline]; ok {
 					pc.Bump() // raises flag in pipelineController to run again whenever it finishes
 				} else {
-					// Initialize op ctx (cancelled at the end of pipelineController.Start(), to avoid leaking
-					// resources), whereas masterClient is passed by the
-					// PPS master and used in case a monitor needs to be spawned for 'pipeline',
-					// whose lifetime is tied to the master rather than this op.
+					// pc's ctx is cancelled at the end of pipelineController.Start(), to avoid leaking resources
```

Comment:
it might make sense to update this comment too—I believe the context is now cancelled in `tryFinish()`